dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/notification/ProgramNotificationTemplateRequestParamsMapper.java

```diff
@@ -0,0 +1,96 @@
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * Neither the name of the HISP project nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.webapi.controller.notification;
+
+import java.util.Objects;
+import lombok.RequiredArgsConstructor;
+import org.hisp.dhis.feedback.BadRequestException;
+import org.hisp.dhis.feedback.ConflictException;
+import org.hisp.dhis.program.notification.NotificationPagingParam;
+import org.hisp.dhis.program.notification.ProgramNotificationTemplateOperationParams;
+import org.springframework.stereotype.Component;
+
+/**
+ * @author Zubair Asghar
+ */
+@Component
+@RequiredArgsConstructor
+public class ProgramNotificationTemplateRequestParamsMapper {
+  public ProgramNotificationTemplateOperationParams map(
+      ProgramNotificationTemplateRequestParams requestParams)
+      throws ConflictException, BadRequestException {
+
+    validateRequestParams(requestParams);
+
+    boolean isPaged = determinePaging(requestParams);
+
+    return ProgramNotificationTemplateOperationParams.builder()
+        .program(requestParams.getProgram())
+        .programStage(requestParams.getProgramStage())
+        .skipPaging(!isPaged)
+        .paged(isPaged)
+        .page(
+            isPaged
+                ? Objects.requireNonNullElse(
+                    requestParams.getPage(), NotificationPagingParam.DEFAULT_PAGE)
+                : null)
+        .pageSize(
+            isPaged
+                ? Objects.requireNonNullElse(
+                    requestParams.getPageSize(), NotificationPagingParam.DEFAULT_PAGE_SIZE)
+                : null)
+        .build();
+  }
+
+  private void validateRequestParams(ProgramNotificationTemplateRequestParams requestParams)
+      throws ConflictException {
+
+    if (requestParams.getProgram() == null && requestParams.getProgramStage() == null) {
+      throw new ConflictException("Program or ProgramStage must be specified.");
```

Comment:
```suggestion
      throw new ConflictException("`program` or `programStage` must be specified.");
```

Reply:
We should refere to the parameters as the client is passing them, so it is `program` and not Program